packages/template-retail-react-app/app/pages/account/payments.test.js

```diff
@@ -82,51 +34,69 @@ const MockedComponent = () => {
-        login: 'jkeane@64labs.com'
-    }
 })
 afterEach(() => {
     localStorage.clear()
-    server.resetHandlers()
 })
-afterAll(() => server.close())
 
 test('Allows customer to add and remove payment methods', async () => {
+    global.server.use(
+        rest.post('*/payment-instruments', (req, res, ctx) => res(ctx.delay(0), ctx.status(200))),
+        rest.get('*/customers/:customerId', (req, res, ctx) =>
+            res(ctx.delay(0), ctx.status(200), ctx.json(mockedRegisteredCustomer))
+        )
+    )
     renderWithProviders(<MockedComponent />)
-    await waitFor(() => expect(screen.getByText('registeredCustomerId')).toBeInTheDocument())
+    await waitFor(() => expect(screen.getByText('customerid')).toBeInTheDocument())
 
-    expect(screen.getByText(/no saved payment methods/i)).toBeInTheDocument()
+    const updatedCustomer = {...mockedRegisteredCustomer}
+    const newPayment = {
+        creationDate: '2021-04-01T14:34:56.000Z',
+        lastModified: '2021-04-01T14:34:56.000Z',
+        paymentBankAccount: {},
+        paymentCard: {
+            cardType: 'Visa',
+            creditCardExpired: false,
+            expirationMonth: 12,
+            expirationYear: 2030,
+            holder: 'Test Customer',
+            maskedNumber: '************1111',
+            numberLastDigits: '1111',
+            validFromMonth: 1,
+            validFromYear: 2020
+        },
+        paymentInstrumentId: 'testcard2',
+        paymentMethodId: 'CREDIT_CARD'
+    }
+    updatedCustomer.paymentInstruments.push(newPayment)
+    global.server.use(
+        rest.get('*/customers/:customerId', (req, res, ctx) =>
+            res(ctx.delay(0), ctx.status(200), ctx.json(updatedCustomer))
+        )
+    )
 
-    // add
     user.click(screen.getByText(/add payment method/i))
     user.type(screen.getByLabelText(/card number/i), '4111111111111111')
     user.type(screen.getByLabelText(/name on card/i), 'Test Customer')
     user.type(screen.getByLabelText(/expiration date/i), '1230')
     user.type(screen.getByLabelText(/security code/i), '555')
+
     user.click(screen.getByText('Save'))
     expect(await screen.findByText('Visa')).toBeInTheDocument()
-    expect(await screen.findByText('•••• 1111')).toBeInTheDocument()
-    expect(await screen.findByText('12/2030')).toBeInTheDocument()
     expect(await screen.findByText('Test Customer')).toBeInTheDocument()
 
+    global.server.use(
+        rest.get('*/customers/:customerId', (req, res, ctx) =>
+            res(ctx.delay(0), ctx.status(200), ctx.json(mockedRegisteredCustomer))
+        )
+    )
     // remove
-    user.click(screen.getByText(/remove/i))
-    expect(await screen.findByText(/no saved payment methods/i)).toBeInTheDocument()
+    user.click(screen.getAllByText(/remove/i)[1])
+    const loadingEl = await screen.getAllByText('Loading...')
+    waitForElementToBeRemoved(loadingEl).then(() =>
+        expect(screen.queryByText('Test Customer')).not.toBeInTheDocument()
+    )
```

Comment:
```suggestion
    await waitForElementToBeRemoved(loadingEl)
    expect(screen.queryByText('Test Customer')).not.toBeInTheDocument()
```
1. Gotta `await` the promise, otherwise jest won't properly report if it fails.
2. If you're using `await` then you don't need the `.then` chain.